src/primitives.rs

```diff
@@ -1,10 +1,167 @@
 //! Contains comparison primitives used to build up the rest of the library.
 
+use core::{
+    cmp::Ordering,
+    convert::{From, TryFrom},
+};
+
+// TODO: turn primitives.rs into
```

Comment:
I can make a branch and do this, if needed. IntelliJ Rust has a nice feature for transforming a single file into a module.